src/main/scala/SerialAdapter.scala

```diff
@@ -213,20 +226,9 @@ trait CanHavePeripherySerialModuleImp extends LazyModuleImp {
     None
   }
 
-  def connectSimSerial() = {
-    val sim = Module(new SimSerial(SERIAL_IF_WIDTH))
-    sim.io.clock := clock
-    sim.io.reset := reset
-    sim.io.serial <> serial.get
-    sim.io.exit
-  }
+  def connectSimSerial() = serial.map { s => SerialAdapter.connectSimSerial(s, clock, reset) }.getOrElse(false.B)
```

Comment:
Why not do the Option[SerialIO] thing like in the block device?

Reply:
Just to clarify, you mean put the `.map` in the `SerialAdapter` object functions? I can do that.

Reply:
Did that